### PR TITLE
Java: Add Zrank and Zrankwithscores command. (Sorted Set Commands)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 1]
```diff
@@ -183,4 +183,48 @@ CompletableFuture<Double> zaddIncr(
...
+    CompletableFuture<Long> zrank(String key, String member);
+
+    /**
+     * Returns the rank of <code>member</code> in the sorted set stored at <code>key</code> with it's score, where scores
+     * are ordered from the lowest to highest.
+     *
+     * @see <a href="https://redis.io/commands/zrank/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member The member whose rank is to be retrieved.
+     * @return An array containing the rank (as Long) and score (as Double) of <code>member</code> in the sorted
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:216]:
```suggestion
     * @return An array containing the rank (as <code>Long</code>) and score (as <code>Double</code>) of <code>member</code> in the sorted
```

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 2]
```diff
@@ -183,4 +183,48 @@ CompletableFuture<Double> zaddIncr(
...
+     *     set.<br>
+     *     If <code>key</code> doesn't exist, or if <code>member</code> is not present in the set,
+     *     null will be returned.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:219]:
```suggestion
     *     <code>null</code> will be returned.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -1241,6 +1242,42 @@ public T zcard(@NonNull String key) {
...
+     * Returns the rank of <code>member</code> in the sorted set stored at <code>key</code> with it's score, where scores
+     * are ordered from the lowest to highest.
+     *
+     * @see <a href="https://redis.io/commands/zrank/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member The member whose rank is to be retrieved.
+     * @return Command Response - An array containing the rank (as Long) and score (as Double) of <code>member</code> in the sorted
+     *     set.<br>
+     *     If <code>key</code> doesn't exist, or if <code>member</code> is not present in the set,
+     *     null will be returned.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1255]:
```suggestion
     *     <code>null</code> will be returned.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -1241,6 +1242,42 @@ public T zcard(@NonNull String key) {
...
+    }
+
+    /**
+     * Returns the rank of <code>member</code> in the sorted set stored at <code>key</code> with it's score, where scores
+     * are ordered from the lowest to highest.
+     *
+     * @see <a href="https://redis.io/commands/zrank/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member The member whose rank is to be retrieved.
+     * @return An array containing the rank (as Long) and score (as Double) of <code>member</code> in the sorted
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1270]:
```suggestion
     * @return An array containing the rank (as <code>Long</code>) and score (as <code>Double</code>) of <code>member</code> in the sorted
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 5]
```diff
@@ -1241,6 +1242,42 @@ public T zcard(@NonNull String key) {
...
+     *     set.<br>
+     *     If <code>key</code> doesn't exist, or if <code>member</code> is not present in the set,
+     *     null will be returned.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1273]:
```suggestion
     *     <code>null</code> will be returned.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 6]
```diff
@@ -1241,6 +1242,42 @@ public T zcard(@NonNull String key) {
...
     }
 
+    /**
+     * Returns the rank of <code>member</code> in the sorted set stored at <code>key</code> with it's score, where scores
+     * are ordered from the lowest to highest.
+     *
+     * @see <a href="https://redis.io/commands/zrank/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member The member whose rank is to be retrieved.
+     * @return Command Response - An array containing the rank (as Long) and score (as Double) of <code>member</code> in the sorted
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1252]: I don't think zrank returns an array.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 7]
```diff
@@ -1241,6 +1242,42 @@ public T zcard(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Returns the rank of <code>member</code> in the sorted set stored at <code>key</code> with it's score, where scores
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1246]: this is different from above

File: java/client/src/main/java/glide/api/models/BaseTransaction.java (hunk 6 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/models/BaseTransaction.java:1252]: True copy paste issue.

**Pull request merged**